### ISSUE TITLE
GPT2 generations with specific words

### ISSUE THREAD

**Issue by prabalbansal** [huggingface/transformers#3910]
# ❓ Questions & Help

## Details
@patrickvonplaten 
Just curious to generate a text with the specified words.
Generate function has a parameter bad_words, instead of removing these, I want to include these in generations. I try to edit in src/transformers/modeling_utils.py line 1194; to 
next_token_logits[batch_idx, banned_tokens[batch_idx]] = -float("inf")
changed to :
next_token_logits[batch_idx, banned_tokens[batch_idx]] = 1
 But this didn't work. Is there is any way we can do this?

**Comment by patrickvonplaten** [comment]
These `next_token_logits[batch_idx, banned_tokens[batch_idx]]` are logits in the `range(-inf, inf)` not probabilities. You can try to set all other logits except your banned_tokens to `-float("inf")`
